packages/api/src/services/prices.ts

```diff
@@ -11,26 +12,84 @@ export default function (config: Config, libs: Libs) {
   assert(coingecko, "requires coingecko library");
   assert(prices[currency], `requires prices.${currency}`);
 
-  async function updatePrice(address: string) {
+  // if we have a new emp address, this will create a new price table structure to store historical price data
```

Comment:
Are we exclusively relying on coingecko to give us the market price of the synthetic tokens?

Are we expecting to use this to do things like compute collateralization ratio, etc?

These seem like shaky assumptions that created problems in the mining scripts. Should we not use our own price feed libraries to get more accurate market and notional prices?
What happens if coingecko doesn't list a token?

Reply:
this is only for collaterals, synthetic tokens is a future pr and will need to use bot price feeds!

Reply:
I see. That makes sense, then. Although it's not entirely clear if CG lists prices for all approved collateral tokens, I think this is probably a good starting point!

As a side note: long term, it might be preferable to use (as a fallback or replacement) 1inch or other DEX aggregator to grab prices (although there are tradeoffs there too).

Reply:
Also, why do we need collateral prices?

Reply:
this is to calculate tvl per emp and over all emps in usd.  a lot of this overlaps with the work chris did with the merkle tree tvl calculations.